reconcile/ocm_machine_pools.py

```diff
@@ -504,19 +546,73 @@ def calculate_diff(
     return diffs, errors
 
 
-def act(dry_run: bool, diffs: Iterable[PoolHandler], ocm_map: OCMMap) -> None:
+def calculate_spec_drift(
+    current_state: Mapping[str, list[AbstractPool]],
+    desired_state: Mapping[str, DesiredMachinePool],
+) -> list[tuple[str, ClusterMachinePoolV1]]:
+    """
+    Finds spec drifts between OCM and app-interface and returns a list of them.
+    """
+    current_machine_pools = {
+        (cluster_name, machine_pool.id): machine_pool
+        for cluster_name, machine_pools in current_state.items()
+        for machine_pool in machine_pools
+    }
+
+    desired_machine_pools = {
+        (desired.cluster_name, desired_machine_pool.q_id): desired_machine_pool
+        for desired in desired_state.values()
+        for desired_machine_pool in desired.pools
+    }
+    spec_drift_updates: list[tuple[str, ClusterMachinePoolV1]] = []
+    for pool_key, current_pool in current_machine_pools.items():
+        desired_pool = desired_machine_pools.get(pool_key)
+        if not desired_pool:
+            continue
+        update = current_pool.get_pool_spec_to_update(desired_pool)
+        if update:
+            cluster_path = desired_state[pool_key[0]].cluster_path
+            spec_drift_updates.append((cluster_path, update))
+    return spec_drift_updates
```

Comment:
Since the return type `list[tuple[str, ClusterMachinePoolV1]]` requires additional transform to `dict[str, list[ClusterMachinePoolV1]]` anyway, how about create it directly here to simplify? Like this:

```python
def _drift_updates(
    current_pools: list[AbstractPool],
    desired: DesiredMachinePool,
) -> list[ClusterMachinePoolV1]:
    current_machine_pools = {pool.id: pool for pool in current_pools}
    return [
        spec
        for pool in desired.pools
        if (current_pool := current_machine_pools.get(pool.q_id))
        and (spec := current_pool.get_pool_spec_to_update(pool)) is not None
    ]

def calculate_spec_drift(
    current_state: Mapping[str, list[AbstractPool]],
    desired_state: Mapping[str, DesiredMachinePool],
) -> list[tuple[str, ClusterMachinePoolV1]]:
    return {
        desired_state[k].cluster_name: updates
        for k in current_state.keys() & desired_state.keys()
        if (updates := _drift_updates(current_state[k], desired_state[k]))
    }
```

Reply:
done. thank you